src/index.ts

```diff
@@ -1,31 +1,75 @@
+import { parseOrbManifest } from "@circleci/circleci-config-parser";
 import { orb } from "@circleci/circleci-config-sdk";
 import { OrbImportManifest } from "@circleci/circleci-config-sdk/dist/src/lib/Orb/types/Orb.types";
 import { fetch } from "undici";
+import { parse } from "yaml";
 import OrbImport = orb.OrbImport;
 
+export interface OrbVersionRef {
+  namespace: string;
+  orb: string;
+  version: string;
+}
+
+export type OrbImportRef = { alias: string } & OrbVersionRef;
+
 /// Downloads the manifest for the given orb and generates the `OrbImport` for use in the SDK
 export async function importOrb({
   alias,
   namespace,
   orb,
   version,
-}: {
-  alias: string;
-  namespace: string;
-  orb: string;
-  version: string;
-}): Promise<OrbImport> {
-  const manifest = await fetchOrbManifest(`${namespace}/${orb}@${version}`);
-  return new OrbImport(alias, namespace, orb, version, "", manifest);
+}: OrbImportRef): Promise<OrbImport> {
+  const source = await fetchOrbSource({ namespace, orb, version });
+  const manifest = await generateOrbManifestFromSource(source);
+  return new OrbImport(alias, namespace, orb, version, undefined, manifest);
 }
 
-async function fetchOrbManifest(orbId: string): Promise<OrbImportManifest> {
-  const endpoint = "https://orb-indexer-proxy.herokuapp.com";
-  const response = await fetch(
-    `${endpoint}/orbs?orb=${encodeURIComponent(orbId)}`
-  );
-  return response.json() as Promise<OrbImportManifest>;
+// Fetches the YAML source for the specified orb
+export async function fetchOrbSource({
+  namespace,
+  orb,
+  version,
+}: OrbVersionRef): Promise<string> {
+  const orbVersionRef = `${namespace}/${orb}@${version}`;
+  const response = await fetch("https://circleci.com/graphql-unstable", {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify({
+      query: `
+        query ($orbVersionRef: String!) {
+          orbVersion(orbVersionRef: $orbVersionRef) {
+            source
+          }
+        }
+      `,
+      variables: { orbVersionRef },
+    }),
+  });
+  return ((await response.json()) as any).data.orbVersion.source;
 }
 
-// TODO: Remove this export once https://github.com/CircleCI-Public/visual-config-editor/issues/296 is fixed
-export * from "./no-parameters";
+// Generates the orb manifest from the orb source YAML
+export function generateOrbManifestFromSource(
+  sourceYaml: string
+): OrbImportManifest {
+  const source = parse(sourceYaml);
+  /* eslint-disable no-restricted-syntax */
+  for (const componentType of ["jobs", "commands", "executors"]) {
+    for (const componentName in source[componentType]) {
+      if (
+        Object.prototype.hasOwnProperty.call(
+          source[componentType],
+          componentName
+        )
+      ) {
+        const parameterList = source[componentType][componentName].parameters;
+        source[componentType][componentName] = parameterList || {};
+      }
+    }
+  }
```

Comment:
You could consider writing this double loop in a more functional programming style. JavaScript was designed to be a functional programming language (and the imperative elements were bolted on because the management wanted it to look like Java).

Going functional is the path of least resistance in JavaScript:

```ts
export function generateOrbManifestFromSource(
  sourceYaml: string
): OrbImportManifest {
  const source = parse(sourceYaml);

  const manifest = Object.keys(source).reduce((rootMap, componentType) => ({
    ...rootMap,
    ...Object.keys(source[componentType]).reduce((compMap, componentName) => {
      const params = compMap[componentName].parameters;
      return {
        ...compMap,
        [componentName]: params ?? {}
      }
    }, {})
  }), {});

return parseOrbManifest(manifest) as OrbImportManifest;
}
```

I would at least consider changing that `if` statement to be more like this:
```
if (source[componentType].hasOwnProperty(componentName) {}
```

which is a lot less dense and more readable.

Reply:
I only added that check because eslint kept telling me to, lol. Not sure I understand what edge case it was worried about either